main.go

```diff
@@ -12,6 +24,125 @@ func init() {
 	_ = flag.Set("logtostderr", "true")
 }
 
+func GetCurrentNode(clientset kubernetes.Clientset) (*corev1.Node, error) {
+	nodeName := os.Getenv("NODE_NAME")
```

Comment:
a little weird having an `os.Getenv` especially if we move this to a package. maybe better to pass a node name to this func and you could setup a package var you set in main and do the empty check there. this would be much simpler too as just...

```
return clientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
```

Reply:
But that will require passing `nodeName` to everywhere. Do you think that would be fine?
```diff
diff --git a/main.go b/main.go
index f403f4c..ae113db 100644
--- a/main.go
+++ b/main.go
@@ -2,7 +2,6 @@ package main
 
 import (
        "context"
-       "errors"
        "flag"
        "fmt"
        "os"
@@ -24,12 +23,7 @@ func init() {
        _ = flag.Set("logtostderr", "true")
 }
 
-func GetCurrentNode(clientset *kubernetes.Clientset) (*corev1.Node, error) {
-       nodeName := os.Getenv("NODE_NAME")
-       if nodeName == "" {
-               return nil, errors.New("Environment variable NODE_NAME is not set")
-       }
-
+func GetCurrentNode(clientset *kubernetes.Clientset, nodeName string) (*corev1.Node, error) {
        node, err := clientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
        if err != nil {
                return nil, fmt.Errorf("failed to get node: %w", err)
@@ -41,12 +35,13 @@ func GetCurrentNode(clientset *kubernetes.Clientset) (*corev1.Node, error) {
 func UpdateNodeLabels(
        clientset *kubernetes.Clientset,
        instanceData *metadata.InstanceData,
+       nodeName string,
 ) error {
        if instanceData == nil {
                return fmt.Errorf("instance data received from Linode metadata service is nil")
        }
 
-       node, err := GetCurrentNode(clientset)
+       node, err := GetCurrentNode(clientset, nodeName)
        if err != nil {
                return fmt.Errorf("failed to get the node: %w", err)
        }
@@ -83,12 +78,17 @@ func GetClientset() (*kubernetes.Clientset, error) {
        return clientset, nil
 }
 
-func StartDecorator(client metadata.Client, clientset *kubernetes.Clientset, interval time.Duration) {
+func StartDecorator(
+       client metadata.Client,
+       clientset *kubernetes.Clientset,
+       interval time.Duration,
+       nodeName string,
+) {
        instanceData, err := client.GetInstance(context.TODO())
        if err != nil {
                klog.Errorf("Failed to get the initial instance data: %s", err.Error())
        } else {
-               err = UpdateNodeLabels(clientset, instanceData)
+               err = UpdateNodeLabels(clientset, instanceData, nodeName)
        }
 
        if err != nil {
@@ -104,7 +104,7 @@ func StartDecorator(client metadata.Client, clientset *kubernetes.Clientset, int
        for {
                select {
                case data := <-instanceWatcher.Updates:
-                       err = UpdateNodeLabels(clientset, data)
+                       err = UpdateNodeLabels(clientset, data, nodeName)
                        if err != nil {
                                klog.Fatal(err)
                        }
@@ -130,7 +130,12 @@ func main() {
                klog.Fatal(err)
        }
 
-       _, err = GetCurrentNode(clientset)
+       nodeName := os.Getenv("NODE_NAME")
+       if nodeName == "" {
+               klog.Fatal("Environment variable NODE_NAME is not set")
+       }
+
+       _, err = GetCurrentNode(clientset, nodeName)
        if err != nil {
                klog.Fatal(err)
        }
@@ -143,5 +148,5 @@ func main() {
                klog.Fatal(err)
        }
 
-       StartDecorator(*client, clientset, *interval)
+       StartDecorator(*client, clientset, *interval, nodeName)
 }
```

Reply:
make it a package var then and just reference it from that, no point in pulling it from env every time

Reply:
Oh yea, that's right. Will do